browser/brave_wallet/brave_wallet_tab_helper.cc

```diff
@@ -0,0 +1,109 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_wallet/brave_wallet_tab_helper.h"
+
+#include <string>
+#include <vector>
+
+#include "brave/browser/ui/views/frame/brave_browser_view.h"
+#include "brave/common/webui_url_constants.h"
+#include "brave/components/brave_wallet/browser/ethereum_permission_utils.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/browser_list.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "components/grit/brave_components_strings.h"
+#include "components/permissions/permission_request.h"
+#include "components/permissions/permission_request_manager.h"
+#include "components/permissions/request_type.h"
+#include "components/sessions/content/session_tab_helper.h"
+#include "content/public/browser/web_contents.h"
+#include "ui/views/view.h"
+
+namespace {
+
+content::WebContents* GetActiveWebContents() {
+  return BrowserList::GetInstance()
+      ->GetLastActive()
+      ->tab_strip_model()
+      ->GetActiveWebContents();
+}
+
+}  // namespace
+
+namespace brave_wallet {
+
+BraveWalletTabHelper::BraveWalletTabHelper(content::WebContents* web_contents)
+    : web_contents_(web_contents) {}
+
+BraveWalletTabHelper::~BraveWalletTabHelper() = default;
+
+void BraveWalletTabHelper::ShowBubble() {
+  // Suppress request if not from active web_contents.
+  if (GetActiveWebContents() != web_contents_)
+    return;
+
+  Browser* browser = chrome::FindBrowserWithWebContents(web_contents_);
+  DCHECK(browser);
+
+  views::View* anchor_view = static_cast<BraveBrowserView*>(browser->window())
+                                 ->GetWalletButtonAnchorView();
+  DCHECK(anchor_view);
+
+  webui_bubble_manager_ = std::make_unique<WebUIBubbleManagerT<WalletPanelUI>>(
+      anchor_view, browser->profile(), GetBubbleURL(),
+      IDS_ACCNAME_BRAVE_WALLET_BUTTON, true);
+
+  webui_bubble_manager_->ShowBubble();
+}
+
+void BraveWalletTabHelper::CloseBubble() {
+  webui_bubble_manager_->CloseBubble();
+}
+
+bool BraveWalletTabHelper::IsShowingBubble() {
+  return webui_bubble_manager_ && webui_bubble_manager_->GetBubbleWidget();
+}
+
```

Comment:
I think we also need to implement `OnVisibilityChanged` to close the bubble when you switch tabs

Reply:
As discussed in DM, tab switch is handled by webui_bubble_manager.